app/models/mixins/file_depot_mixin.rb

```diff
@@ -12,7 +12,7 @@ module FileDepotMixin
 
   included do
     include AuthenticationMixin
-    before_save :verify_uri_prefix_before_save
+    before_validation :verify_uri_prefix_before_save, :validate_depot_credentials
```

Comment:
Why use `before_validation` here? As far as I know that's only if you want to massage the data in some way before saving it, but that's not happening in `validate_depot_credentials`.